### PR TITLE
Enable https for proxy offloading

### PR DESCRIPTION
**Pull request by ElGranLoky**
If we install ospos in docker behind a proxy that performs ssloffloading, we can enable the url generated to be https instead of http, by activating the environment variable FORCE_HTTPS = 1.

### CODE REVIEW

File: application/config/config.php [hunk 1]
```diff
@@ -65,9 +65,15 @@
...
 |
 */
-$config['base_url'] = (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] == 'on') ? 'https' : 'http';
-$config['base_url'] .= '://' . $_SERVER['HTTP_HOST'];
-$config['base_url'] .= str_replace(basename($_SERVER['SCRIPT_NAME']), '', $_SERVER['SCRIPT_NAME']);
+if (isset($_ENV['FORCE_HTTPS'])) {
+  $config['base_url'] = 'https';
+  $config['base_url'] .= '://' . $_SERVER['HTTP_HOST'];
+  $config['base_url'] .= str_replace(basename($_SERVER['SCRIPT_NAME']), '', $_SERVER['SCRIPT_NAME']);
+} else {
```
Comment (jekkos) [inline application/config/config.php]: two lines of code duplication here, why not add the last two lines after the if-clause?

Comment (mckaygerhard) [reply application/config/config.php]: better put:

``` php
if (isset($_ENV['FORCE_HTTPS']))
$_SERVER['HTTPS']  = 'on'
```

that's all

**Comment by jekkos** [comment]
@mckaygerhard that's not completely according to the coding indentation standards we adhere. Should use k&r style indentation for if clauses.

Might be nitpicking for this small change but take care of that in the furutre

**Comment by mckaygerhard** [comment]
@jekkos  all the pull request are editable by the repository owers, and editing that code to corrrect are easy due are only one line.. so why dont just only edit that great idea.. 

@jekkos  i only aprobe the optimization of code of course, do you note that the wiki about codign are not consistent? due that are not "final" and currently anybody (rather than me) take care of a good explained wiki, it's common and normal that kind of mistakes... so that's why aprobe that code.. 

A GOOD FILOSOFY: for more additions of contributors few lines can be corrected by owers, if are more complicated or more thant 100 lines the contributor must correct and made requested changes..

the secret for a popular project its that will be easy to made contributions.. not complicated!

**Comment by ElGranLoky** [comment]
Sorry @jekkos I did not know anything about the k&r policy and perform the code tests by making the modification of @mckaygerhard exactly as it was written. When checking it gave me the following errors: 

3/3/2018 13:55:49[Sat Mar 03 12:55:49.295799 2018] [:error] [pid 21] [client 10.42.89.166:41268] PHP Parse error:  syntax error, unexpected variable (T_VARIABLE) in /app/application/config/config.php on line 74
3/3/2018 13:55:49[Sat Mar 03 12:55:49.295862 2018] [:error] [pid 21] [client 10.42.89.166:41268] PHP Fatal error:  Exception thrown without a stack frame in Unknown on line 0
3/3/2018 13:55:49[Sat Mar 03 12:55:49.296201 2018] [:error] [pid 21] [client 10.42.89.166:41268] PHP Parse error:  syntax error, unexpected '$config' (T_VARIABLE) in /app/application/config/config.php on line 74
3/3/2018 13:55:49[Sat Mar 03 12:55:49.296230 2018] [:error] [pid 21] [client 10.42.89.166:41268] PHP Fatal error:  Exception thrown without a stack frame in Unknown on line 0
3/3/2018 13:55:4910.42.89.166 - - [03/Mar/2018:12:55:49 +0000] "GET / HTTP/1.1" 500 211 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/63.0.3239.132 Safari/537.36"

so I chose to write it on a line adding the semicolon, looking at other examples.

I just made the changes respecting the KB policy, Rules are to fulfill them 💯

File: application/config/config.php [hunk 2]
```diff
@@ -65,6 +65,9 @@
 | a PHP script and you can easily do that on your own.
 |
 */
+if (isset($_ENV['FORCE_HTTPS']))
+    $_SERVER['HTTPS'] = 'on';
+
 $config['base_url'] = (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] == 'on') ? 'https' : 'http';
```
Comment (daN4cat) [inline application/config/config.php]: just make it removing the previous added lines:

 $config['base_url'] = ((isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] == 'on') || isset($_ENV['FORCE_HTTPS'])) ? 'https' : 'http';

File: application/config/config.php [hunk 3]
```diff
@@ -65,10 +65,8 @@
 | a PHP script and you can easily do that on your own.
 |
 */
-if (isset($_ENV['FORCE_HTTPS']))
-    $_SERVER['HTTPS'] = 'on';
 
-$config['base_url'] = (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] == 'on') ? 'https' : 'http';
+$config['base_url'] = (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] == 'on') || isset($_ENV['FORCE_HTTPS'])) ? 'https' : 'http';
```
Comment (daN4cat) [inline application/config/config.php]: it's missing one ( at the beginning of the line...

Comment (daN4cat) [reply application/config/config.php]: you need to test your changes before submitting a PR.

**Comment by daN4cat** [comment]
I fixed and committed to master.

**Comment by ElGranLoky** [comment]
thanks

**Pull request closed without merging**